src/reconnector.ts

```diff
@@ -47,7 +47,7 @@ export class Reconnector {
         return;
       }
 
-      if (this.client._stanzaio.transport?.hasStream || this.client.connecting) {
+      if (this.client._stanzaio.transport?.hasStream) {
```

Comment:
The issue was `this.client.connecting` was set to `true` but stanza tore down the actual websocket. So we were sitting in a state where the reconnector _thought_ we were reconnecting but really we were just waiting for backoff attempts to run out before trying to hard reconnect. 

With this change, it will not drop into this `if` block and will go on to line 58 where stanza tries to create another WS with the existing channel.